server/container/guice/postgres-common/src/main/java/org/apache/james/modules/data/PostgresCommonModule.java

```diff
@@ -91,9 +88,15 @@ JamesPostgresConnectionFactory provideJamesPostgresConnectionFactory(PostgresCon
     @Named(JamesPostgresConnectionFactory.NON_RLS_INJECT)
     @Singleton
     JamesPostgresConnectionFactory provideJamesPostgresConnectionFactoryWithRLSBypass(PostgresConfiguration postgresConfiguration,
+                                                                                      JamesPostgresConnectionFactory jamesPostgresConnectionFactory,
                                                                                       @Named(JamesPostgresConnectionFactory.NON_RLS_INJECT) ConnectionFactory connectionFactory) {
-        LOGGER.info("Implementation for PostgresSQL connection factory: {}", SinglePostgresConnectionFactory.class.getName());
-        return new SinglePostgresConnectionFactory(Mono.from(connectionFactory.create()).block());
+        if (!postgresConfiguration.rowLevelSecurityEnabled()) {
+            return jamesPostgresConnectionFactory;
+        }
+        return new PoolBackedPostgresConnectionFactory(DISABLED_ROW_LEVEL_SECURITY,
+            postgresConfiguration.nonRLSPoolInitialSize(),
+            postgresConfiguration.nonRLSPoolMaxSize(),
+            connectionFactory);
```

Comment:
If i understand here, you return the normal pool if rls was disabled, but returning an extra pool if rls was enabled with non rls?

Reply:
when we use rls-bypass crendentials, there is no need to set domain for connection (the first para of PoolBackedPostgresConnectionFactory constructor is used to check whether setting domain is needed)